### PR TITLE
Issue #5422: Most of readDataBlob_cancel events do not have error message

### PR DESCRIPTION
**Pull request by vladsud**
Also stop casting to any to add properties, or using spread operator (later does not really work - none of properties would get logged as we would lose getTelemetryProperties() method

### CODE REVIEW

File: packages/utils/telemetry-utils/src/logger.ts [hunk 1]
```diff
@@ -498,6 +498,12 @@ export class LoggingError extends Error {
         props?: ITelemetryProperties,
     ) {
         super(message);
+        if (props) {
+            this.addProperties(props);
+        }
+    }
+
+    public addProperties(props: ITelemetryProperties) {
```
Comment (markfields) [inline packages/utils/telemetry-utils/src/logger.ts:506]: Yay!  I have this same change pending [here](https://github.com/microsoft/FluidFramework/pull/5308/files#diff-a8e8af708cd24804ad20446f3a666813d33871f415efed5b116b64a381f93298R592-R598), but I called it `addTelemetryProperties` for symmetry with `getTelemetryProperties`, what do you think?

File: packages/utils/telemetry-utils/src/logger.ts [hunk 2]
```diff
@@ -498,6 +498,12 @@ export class LoggingError extends Error {
         props?: ITelemetryProperties,
     ) {
         super(message);
+        if (props) {
```
Comment (markfields) [inline packages/utils/telemetry-utils/src/logger.ts:501]: unnecessary 🤷‍♂️

File: packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts [hunk 3]
```diff
@@ -183,19 +183,18 @@ export function createOdspNetworkError(
 
     error.online = OnlineStatus[isOnline()];
 
-    const errorAsAny = error as any;
-
-    errorAsAny.response = responseText;
+    const props: ITelemetryProperties = { response: responseText };
```
Comment (markfields) [inline packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts:186]: Ok I'm having a major double-take here - I did this _exact_ change in a branch at one point but dropped it...  Great minds think alike I guess!

File: packages/drivers/odsp-driver/src/epochTracker.ts [hunk 4]
```diff
@@ -214,23 +214,23 @@ export class EpochTracker {
                 // This will only throw if it is an epoch error.
                 this.checkForEpochErrorCore(epochFromResponse, error.errorMessage);
             } catch (epochError) {
-                const err = {
-                    ...epochError,
+                assert(epochError instanceof LoggingError);
```
Comment (markfields) [inline packages/drivers/odsp-driver/src/epochTracker.ts:217]: Add a small message please :)

File: packages/drivers/odsp-driver/src/epochTracker.ts [hunk 5]
```diff
@@ -214,23 +214,23 @@ export class EpochTracker {
...
                 assert(!!this.fileEntry, "File Entry should be set to clear the cached entries!!");
                 // If the epoch mismatches, then clear all entries for such file entry from cache.
                 await this.persistedCache.removeEntries(this.fileEntry);
                 throw epochError;
             }
-            // If it was categorised as epoch error but the epoch returned in response matches with the client epoch
+            // If it was categorized as epoch error but the epoch returned in response matches with the client epoch
             // then it was coherency 409, so rethrow it as throttling error so that it can retried. Default throttling
             // time is 1s.
             this.logger.sendErrorEvent({ eventName: "Coherency409" }, error);
```
Comment (markfields) [inline packages/drivers/odsp-driver/src/epochTracker.ts:232]: Is this a helpful log, or noisy/duplicate of the below thrown error which will presumably be logged later as well?

**Review by markfields — APPROVED** [review]
🎂 LGTM

File: packages/utils/telemetry-utils/src/logger.ts (hunk 2 above)
Comment (vladsud) [reply packages/utils/telemetry-utils/src/logger.ts:501]: I prefer stronger typing (changing addTelemetryProperties to not accept undefined).

File: packages/drivers/odsp-driver/src/epochTracker.ts (hunk 5 above)
Comment (vladsud) [reply packages/drivers/odsp-driver/src/epochTracker.ts:232]: So far it as been helpful, though in future we might decide to drop it. Ideally it should not happen that often (due to server having its own retry logic) and presence of it in logs immediately tells us something is off

File: packages/drivers/odsp-driver/src/epochTracker.ts (hunk 4 above)
Comment (vladsud) [reply packages/drivers/odsp-driver/src/epochTracker.ts:217]: Removing static debug strings was proven to be # 1 way to reduce binaries in devmain. Our bundles are not different. There is no need to sprinkle messages all over the place, it's similar to commenting - they are useful here and there, but if you have to comment every line of code, that likely means code can be improved to be more readable.
I'd rather see Tony's work completing and inserting tags all over place, and messages (converted to comments) being optional.

**Pull request merged**